profile/src/wallet_kit_common/types/keys/secp256k1/public_key.rs

```diff
@@ -0,0 +1,383 @@
+use crate::prelude::*;
+
+use bip32::secp256k1::PublicKey as BIP32Secp256k1PublicKey;
+use radix_engine_common::crypto::{
+    Hash, IsHash, Secp256k1PublicKey as EngineSecp256k1PublicKey,
+};
+use transaction::{
+    signing::secp256k1::Secp256k1Signature, validation::verify_secp256k1,
+};
+
+/// A `secp256k1` public key used to verify cryptographic signatures (ECDSA signatures).
+#[serde_as]
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    derive_more::Debug,
+    uniffi::Record,
+)]
+#[serde(transparent)]
+#[debug("{}", self.to_hex())]
+pub struct Secp256k1PublicKey {
+    #[serde_as(as = "serde_with::hex::Hex")]
+    value: Vec<u8>, // FIXME: change to either `radix_engine_common::crypto::Secp256k1PublicKey` or `bip32::secp256k1::PublicKey` once we have proper UniFFI lift/lower/UniffiCustomTypeConverter
```

Comment:
FIXME: check again....

Reply:
fixed in https://github.com/Sajjon/RadixWalletKit/pull/28/commits/9a2e8b7dcb7a5d92da7704a60f39d20aef6a3602 the type now wraps a RET public key, instead of inner type `Vec<u8>` (which just was placeholder before I knew how to write UniFFI converters...)